docs/error_codes.md

```diff
@@ -1,42 +1,103 @@
+SafeTxGas is too high or gas limit is too low to execute the transaction, keep in mind that we require some gas, at least `2500` is needed for events and at least `500` to perform code until execution.  
+
+- **GS011: Could not pay gas costs with ether**   
+There are not enough funds of `ether` to execute the transaction.   
+Make sure that you have enough funds of `ether` or choose other token with enough funds setting it in `gasToken`.  
+
+- **GS012: Could not pay gas costs with token**  
+There are not enough funds of the token choosen on `gasToken` to execute the transaction.      
+Make sure you have enough of the gasToken or set it to `address(0)` if you want use ether.  
+
+- **GS013: Safe transaction failed when gasPrice and safeTxGas were 0**  
+Transaction is not succesful, this could be for due to multiple reasons.
 
 ### General signature validation related
-- `GS020`: `Signatures data too short`
-- `GS021`: `Invalid contract signature location: inside static part`
-- `GS022`: `Invalid contract signature location: length not present`
-- `GS023`: `Invalid contract signature location: data not complete`
-- `GS024`: `Invalid contract signature provided`
-- `GS025`: `Hash has not been approved`
-- `GS026`: `Invalid owner provided`
+- **GS020: Signatures data too short**  
+There are less signatures than the owners threshold.  
+Provide as many as signatures as the owners threshold.  
+
+- **GS021: Invalid contract signature location:** inside static part  
+Wrong contract `v=0` signature because `s` value is pointing inside the static part, instead to dynamic part (should point to the corresponding data signature).   
+Review the value of `s` to point to the beginning of the correct signature.  
+More information about signatures:  https://docs.gnosis-safe.io/contracts/signatures  
+
+- **GS022: Invalid contract signature location:** length not present  
+Wrong contract `v=0` signature because `s` value is greater than the last position of signatures (it's pointing to empty value).   
+Review `s` value points to the correct data signature position or add the missing data signature.   
+ 
+- **GS023: Invalid contract signature location:** data not complete  
+Wrong contract `v=0` signature because `startingPosition + contractSignatureLen` is out of bounds.   
+
+- **GS024: Invalid contract signature provided**  
+The EIP1271 signature provided is wrong.  
+If you don't want to use this type of signature review the `v` value (0 for a contract signature).   
+The `hash` to generate the `signature` must be calculated by the account address provided in `r` value.  
+More information about signatures:  https://docs.gnosis-safe.io/contracts/signatures    
+
+- **GS025: Hash has not been approved**  
+The owner provided on 'r' has not pre-approved the safeTxHash.     
+To pre-approve the safeTxHash call `approveHash` with the safeTxHash calculated by the owner.   
+
+- **GS026: Invalid owner provided**   
```

Comment:
We more info here. This error can happen also if the nonce has changed and therefore the Safe transaction hash is different than expected.